tests/expressions/tests.py

```diff
@@ -1901,6 +1996,7 @@ def test_expressions(self):
         )
         self.assertEqual(repr(Col('alias', 'field')), "Col(alias, field)")
         self.assertEqual(repr(F('published')), "F(published)")
+        self.assertEqual(repr(F('published')[0:2]), "SliceableF(published)")
```

Comment:
Perhaps it's an additional burden that can be deferred for another time, but IMHO it'd be  _cool_ if the `__repr__` included either the subscript object or the final attributes it leads to (the latter would require it be `<SliceableF ...>` though, I guess).

I only bring it up because you specifically asserted it though; it's by no means a requirement given there's already an _acceptable_ repr for it.

Reply:
I think something like the following (untested) code would do?

```python
def __repr__(self):
    start = self.start - 1
    stop = None if self.length is None else start + self.length
    subscript = slice(start, stop)
    return f'{self.__class__.__qualname__}({self.name!r}, {subscript!r})'
```